src/senselab/utils/interfaces/analyses.py

```diff
@@ -0,0 +1,68 @@
+"""Base class for text analysis pipelines."""
+
+import string
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional
+
+from senselab.utils.data_structures.device import DeviceType
+from senselab.utils.model_utils import BaseModelSourceUtils
+
+
+class BaseAnalysis(ABC):
```

Comment:
this is an interesting interface. some thoughts:
- I am curious to see how it generalizes on other tasks
- it may be good to include a validate_output method as well